### PR TITLE
[SPARK-7844] [MLlib] Fix broken tests in KernelDensity

### PR DESCRIPTION
**Pull request by MechCoder**
The densities in KernelDensity are scaled down by
(number of parallel processes X number of points). It should be just no.of samples. This results in broken tests in KernelDensitySuite which haven't been tested properly.

### CODE REVIEW

**Comment by MechCoder** [comment]
Note: In master

    import org.apache.commons.math3.distribution.NormalDistribution
    import org.apache.spark.mllib.stat.KernelDensity

    val rdd = sc.parallelize(Array(5.0))
    val evaluationPoints = Array(5.0, 6.0)
    val densities = new KernelDensity().setSample(rdd).setBandwidth(3.0).estimate(evaluationPoints)
    val normal = new NormalDistribution(5.0, 3.0)

    densities(0) - normal.density(5.0)
    res1: Double = -0.06649038006690546

    val rdd = sc.parallelize(Array(5.0, 10.0))
    val evaluationPoints = Array(5.0, 6.0)
    val densities = new KernelDensity().setSample(rdd).setBandwidth(3.0).estimate(evaluationPoints)
    val normal1 = new NormalDistribution(5.0, 3.0)
    val normal2 = new NormalDistribution(10.0, 3.0)

    
    densities(0) - (normal1.density(5.0) + normal2.density(5.0)) / 2
    res2: Double = -0.04153495159951512

Hence the tests pass

cc @jkbradley @mengxr

File: mllib/src/main/scala/org/apache/spark/mllib/stat/KernelDensity.scala [hunk 1]
```diff
@@ -86,20 +86,20 @@ class KernelDensity extends Serializable {
...
+        x
       },
       (x, y) => {
-        blas.daxpy(n, 1.0, y._1, 1, x._1, 1)
-        (x._1, x._2 + y._2)
+        blas.daxpy(n, 1.0, y, 1, x, 1)
+        x
       })
-    blas.dscal(n, 1.0 / count, densities, 1)
+    blas.dscal(n, 1.0 / sample.count, densities, 1)
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/stat/KernelDensity.scala]: Please aggregate densities and count in a single pass.

**Comment by MechCoder** [comment]
@mengxr fixed!

**Comment by MechCoder** [comment]
Also, we might probably want to generalize this n dimensions (Some code can be borrowed from the GaussianMixture )  and add a user guide. Is anyone actively working on these?

File: mllib/src/test/scala/org/apache/spark/mllib/stat/KernelDensitySuite.scala [hunk 2]
```diff
@@ -29,8 +29,8 @@ class KernelDensitySuite extends FunSuite with MLlibTestSparkContext {
     val densities = new KernelDensity().setSample(rdd).setBandwidth(3.0).estimate(evaluationPoints)
     val normal = new NormalDistribution(5.0, 3.0)
     val acceptableErr = 1e-6
-    assert(densities(0) - normal.density(5.0) < acceptableErr)
-    assert(densities(0) - normal.density(6.0) < acceptableErr)
+    assert(Math.abs(densities(0) - normal.density(5.0)) < acceptableErr)
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/mllib/stat/KernelDensitySuite.scala]: minor: `Math` is deprecated. Use `math` instead.

**Comment by mengxr** [comment]
Also ping @sryza for verification.

**Comment by mengxr** [comment]
Merged into master and branch-1.4. Please update #6387 with a doctest. Thanks!

**Pull request closed without merging**